internal/controllers/ipaddressclaim.go

```diff
@@ -368,3 +448,59 @@ func (h *IPAddressClaimHandler) getClient() (pcclient.Client, error) {
 
 	return c, nil
 }
+
+func ensureRequestIDAnnotation(
+	ctx context.Context,
+	obj ctrlclient.Object,
+	requestAnnotationKey string,
+	cl ctrlclient.Client,
+) (string, error) {
+	// If the claim does not have a request ID annotation set, then we set one
+	// here. This is used so we can send the same reservation request and receive the same task ID back.
+	reqID, found := obj.GetAnnotations()[requestAnnotationKey]
+	if !found {
+		claimPatch := ctrlclient.MergeFrom(obj.DeepCopyObject().(ctrlclient.Object))
+
+		reqUUID, err := uuid.NewV7()
```

Comment:
I don't think we really need time ordering bits but works for now

Reply:
v7 UUIDs are now the preferred format generally as they provide better guarantees of uniqueness, etc.